sources/web/datalab/server.ts

```diff
@@ -299,6 +299,15 @@ function stopVmHandler(request: http.ServerRequest, response: http.ServerRespons
 }
 
 function socketHandler(request: http.ServerRequest, socket: net.Socket, head: Buffer) {
+  // Websocket requests aren't CORS-checked by the browser. Reject any CORS
+  // requests here by checking their host vs origin request headers.
+  if (request.headers.origin &&
+    request.headers.origin.replace(/^.*\/\//, '') !== request.headers.host) {
```

Comment:
I can't understand what this replacement is doing. Can you either document this or rework it to be easier to interpret?

Reply:
Ah, forgot to add the comment. Done.